packages/backend/src/ee/services/ScimService/ScimService.ts

```diff
@@ -525,6 +531,75 @@ export class ScimService extends BaseService {
                 );
             }
 
+            // If setting user to inactive, drop org role to MEMBER and remove project roles
+            if (user.active === false) {
```

Comment:
Should these DB operations be in a transaction? What would the effect be if some operations failed? There doesn't seem to be any indication to admins that trigger this function that any part has failed.  

Also, would it be worth pulling this logic into its own function?